website/upgrade-guides/0-13.html.markdown

```diff
@@ -0,0 +1,367 @@
+[Provider Requirements](/docs/configuration/provider-requirements.html).
+That page also includes some guidance on how to write provider dependencies
+for a module that must remain compatible with both Terraform v0.12 and
+Terraform v0.13; the `terraform 0.13upgrade` result includes a conservative
+version constraint for Terraform v0.13 or later, which you can weaken to
+`>= 0.12.26` if you follow the guidelines in
+[v0.12-Compatible Provider Requirements](/docs/configuration/provider-requirements.html#v012-compatible-provider-requirements).
+
+After you've added explicit provider source addresses to your configuration,
+run `terraform init` again to re-run the provider installer.
+
+-> **Action:** Either run `terraform 0.13upgrade` for your module, or manually update the provider declarations to use explicit source addresses.
+
+## New Filesystem Layout for Local Copies of Providers
+
+As part of introducing the hierarchical provider namespace discussed in the
+previous section, Terraform v0.13 also introduces a new hierarchical directory
+structure for manually-installed providers in the local filesystem.
+
+If you use local copies of official providers or if you use custom in-house
+providers that you have installed manually, you will need to adjust your local
+directories to use the new directory structure.
+
+The previous layout was a single directory per target platform containing
+various executable files named with the prefix `terraform-provider`, like
+`linux_amd64/terraform-provider-google_v2.0.0`. The new expected location for the
+Google Cloud Platform provider for that target platform within one of the local
+search directories would be the following:
+
+```
+registry.terraform.io/hashicorp/google/2.0.0/linux_amd64/terraform-provider-google_v2.0.0
+```
+
+The `registry.terraform.io` above is the hostname of the registry considered
+to be the origin for this provider. The provider source address
+`hashicorp/google` is a shorthand for `registry.terraform.io/hashicorp/google`,
+and the full, explicit form is required for a local directory.
+
+Note that the version number given as a directory name must be written _without_
+the "v" prefix that tends to be included when a version number is used as part
+of a git branch name. If you include that prefix, Terraform will not recognize
+the directory as containing provider packages.
+
+As before, the recommended default location for locally-installed providers
+is one of the following, depending on which operating system you are running
+Terraform under:
+
+* Windows: `%APPDATA%\terraform.d\plugins`
+* All other systems: `~/.terraform.d/plugins`
+
+Terraform v0.13 introduces some additional options for customizing where
+Terraform looks for providers in the local filesystem. For more information on
+those new options, see [Provider Installation](/docs/commands/cli-config.html#provider-installation).
+
+If you use only providers that are automatically installable from Terraform
```

Comment:
Maybe worth noting init will re-download all required providers so no action is _required_?

Reply:
I had tried to keep the main imperatives in the "Action:" callouts, and the one associated with this does include the precondition "If you use local copies of official providers rather than installing them automatically from Terraform Registry", but I agree that the paragraph you called out here does seem to imply that _everyone_ should use `terraform providers mirror`, which was not my intent. I've added some additional words to this sentence to hopefully make it clearer that this paragraph is talking only to folks who want to make local mirrors of official providers.